core/common/libp2p/stream_proxy.hpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <libp2p/connection/stream.hpp>
+
+namespace libp2p::connection {
+  struct StreamProxy : Stream {
```

Comment:
`StreamProxy` is the same with `Stream`. remove?

Reply:
> `StreamProxy` _wrapper base_
without this each wrapper implementation needs to implement all methods, even if it wants to override just one